content/deyemiobaa/building-a-todolist-with-rails.md

```diff
@@ -0,0 +1,351 @@
+
+```ruby
+Rails.application.routes.draw do
+  resources :todos, only: [:index, :create]
+end
+```
+
+## Understanding the Rails routing system
+
+The `resources` method in the code block above creates a set of routes for a given resource. The `only` option specifies the controller methods to create routes for. In our case, we only want to create routes for the `index` and `create` methods. The `resources` method creates the following routes.
+
+| HTTP Verb | Path   | Controller#Action | Used for                    |
+|-----------|--------|-------------------|-----------------------------|
+| GET       | /todos | todos#index       | display a list of all todos |
+| POST      | /todos | todos#create      | create a new todo           |
+
+Without passing the `only` option, the `resources` method would create routes for all the controller methods. The `resources` method creates the following routes.
+
+| HTTP Verb | Path   | Controller#Action | Used for                    |
+|-----------|--------|-------------------|-----------------------------|
+| GET       | /todos | todos#index       | display a list of all todos |
+| GET       | /todos/new | todos#new | return an HTML form for creating a new todo |
+| POST      | /todos | todos#create      | create a new todo           |
+| GET       | /todos/:id | todos#show | display a specific todo |
+| GET       | /todos/:id/edit | todos#edit | return an HTML form for editing a todo |
+| PATCH/PUT | /todos/:id | todos#update | update a specific todo |
+| DELETE    | /todos/:id | todos#destroy | delete a specific todo |
+
+## Creating a new todo item
+
+Now that we have our routes setup, we need add the corresponding method in our controller with some logic to handle the form submission. Open the `app/controllers/todos_controller.rb` file and replace what you have with the following code.
+
+```ruby
+class TodosController < ApplicationController
+  def index
+    @todo = Todo.new
+  end
+
+  def create
+    @todo = Todo.create(description: params[:todo][:description])
+    if @todo.valid?
+      render :index
+    else
+      render :index
+    end
+  end
+end
+```
+
+The `create` method creates a new instance of the Todo model and saves it to the database. The `description` attribute is set to the value of the `description` field in the form. The `params` method returns a hash of the form data. The hash looks like this.
+
+```ruby
+{
+  todo: {
+    description: "new todo"
+  }
+}
+```
+
+The `params[:todo][:description]` syntax gets the value of the hash. The `valid?` method checks if the model is valid. If the model is valid it renders the `index` view and show the list of todos. If the model is not valid, it just renders the `index` view. In a real application, we would want to display the errors to the user.
+
+You can submit the form multiple time to have enough data to test the table. The table will be empty for now. We'll add the table in the next section.
```

Comment:
At this point I can't start the server. It gives me an error on the routes.rb file. It is giving a syntax error on the square bracket of `resources :todos, only [:index, :create]`

I'm going to stop here for now since I can no longer follow along with the tutorial. I'll continue when the prior issues have been dealt with.

Reply:
You're missing a colon after `only`. Should be `only:`

Reply:
That fixed it.